src/draw/vg_lite/lv_vg_lite_utils.c

```diff
@@ -780,6 +780,28 @@ void lv_vg_lite_image_matrix(vg_lite_matrix_t * matrix, int32_t x, int32_t y, co
     }
 }
 
+vg_lite_color_t lv_vg_lite_image_recolor(vg_lite_buffer_t * buffer, const lv_draw_image_dsc_t * dsc)
+{
+    LV_ASSERT_NULL(buffer);
+    LV_ASSERT_NULL(dsc);
+
+    if((buffer->format == VG_LITE_A4 || buffer->format == VG_LITE_A8) || dsc->recolor_opa > LV_OPA_TRANSP) {
+        /* alpha image and image recolor */
+        buffer->image_mode = VG_LITE_MULTIPLY_IMAGE_MODE;
+        return lv_vg_lite_color(dsc->recolor, LV_OPA_MIX2(dsc->opa, dsc->recolor_opa), true);
+    }
+
+    if(dsc->opa < LV_OPA_COVER) {
+        /* normal image opa */
+        buffer->image_mode = VG_LITE_MULTIPLY_IMAGE_MODE;
+        vg_lite_color_t color;
+        lv_memset(&color, dsc->opa, sizeof(color));
```

Comment:
Why do you set reg, green and blue to opa? My intuition says that `recolor` and `recolor_opa` should be used here.

Reply:
No, in `VG_LITE_MULTIPLY_IMAGE_MODE`, the GPU multiplies the RGB component of each pixel of the image with the RGB component of `color` to achieve transparency adjustment, which is always used.